### PR TITLE
feat(breaking): update to Manopt 0.5

### PR DESCRIPTION
**Pull request by Nimrais**

### CODE REVIEW

File: test/projection/projected_to_tests.jl [hunk 1]
```diff
@@ -518,7 +518,7 @@ end
         nothing,
     )
     initialpoint = rand(manifold)
-    direction = MomentumGradient(manifold, initialpoint)
+    direction = Manopt.MomentumGradientRule(manifold; p=initialpoint)
```
Comment (bvdmitri) [inline test/projection/projected_to_tests.jl:521]: I think the new Manopt doesn't require `manifold` as an argument? I think we can use the better improved API

**Review by bvdmitri — CHANGES_REQUESTED** [review]
I would use new API from Manopt where we no longer need to pass manifold as an argument + tests are failing

**Comment by Nimrais** [comment]
> I would use new API from Manopt where we no longer need to pass manifold as an argument + tests are failing

It's magically fixed in 1.11.1, it's smt inside Manopt.jl they have some type instability.

File: test/projection/projected_to_tests.jl [hunk 2]
```diff
@@ -557,5 +557,6 @@ end
 
     @test approximated isa MvNormalMeanCovariance
     @test kldivergence(approximated, true_dist) < 0.01  # Ensure good approximation
-    @test projection.parameters.direction isa MomentumGradient
+    @show typeof(projection.parameters.direction)
```
Comment (bvdmitri) [inline test/projection/projected_to_tests.jl:560]:
```suggestion
```

**Review by bvdmitri — APPROVED** [review]

**Pull request merged**